app/src/main/java/org/sopt/sample/presentation/home/diffUtilAdapter/HomeRepoListAdapter.kt

```diff
@@ -0,0 +1,102 @@
+package org.sopt.sample.presentation.home.diffUtilAdapter
+
+import android.content.Context
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import org.sopt.sample.R
+import org.sopt.sample.databinding.HomeRepoContentItemViewBinding
+import org.sopt.sample.databinding.HomeRepoTitleItemViewBinding
+import org.sopt.sample.presentation.home.data.HomeRecycleData
+import org.sopt.sample.presentation.home.data.HomeRecycleData.HomeConst.REPO_CONTENT_TYPE
+import org.sopt.sample.presentation.home.data.HomeRecycleData.HomeConst.REPO_TITLE_TYPE
+import org.sopt.sample.presentation.home.data.HomeRepoContentData
+import org.sopt.sample.presentation.home.viewholder.HomeViewHolder
+import org.sopt.sample.presentation.home.viewholder.RepoContentViewHolder
+import org.sopt.sample.presentation.home.viewholder.RepoTitleViewHolder
+
+
+class HomeRepoListAdapter(context: Context) :
+    ListAdapter<HomeRecycleData, HomeViewHolder>(HomeRepoDiffUtilItemCallback()) {
+    private val inflater by lazy { LayoutInflater.from(context) } //by laze : 초기화를 최대한 늦추는 효과
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HomeViewHolder {
+        return when (viewType) {
+            REPO_TITLE_TYPE -> RepoTitleViewHolder(
+                HomeRepoTitleItemViewBinding.inflate(
+                    inflater,
+                    parent,
+                    false
+                )
+            )
+            REPO_CONTENT_TYPE -> RepoContentViewHolder(
+                HomeRepoContentItemViewBinding.inflate(
+                    inflater,
+                    parent,
+                    false
+                )
+            )
+            else -> {
+                throw java.lang.IllegalArgumentException()
```

Comment:
```suggestion
                throw IllegalArgumentException()
```
kotlin exception을 활용하면 좋을 것 같아요!